### PR TITLE
chore(python): Switch over some of the custom Python date/time conversions to native PyO3 conversions

### LINKED ISSUE #16158: Switch to PyO3 0.21's native datetime support, now available with abi3
Per @stinodego:

PyO3 has support for the Python `datetime` module:
https://pyo3.rs/main/doc/pyo3/types/struct.pydatetime

However, this was never available under the `abi3` feature flag.

So we have a bunch of workarounds in place to materialize/read `datetime` objects. Specially, we have defined Python functions here:
https://github.com/pola-rs/polars/blob/main/py-polars/polars/_utils/convert.py

...that are called from Rust, for example here:
https://github.com/pola-rs/polars/blob/8ef2e212cf52c318e18d12d38b02c3ad0a918ab2/py-polars/src/conversion/any_value.rs#L61-L64

The 0.21 release notes mention:

> Extended chrono / datetime conversions, including support for the abi3 feature and the chrono-tz crate

So I am expecting that we can now take advantage of native PyO3 datetime support. And I expect a good performance boost + code cleanup to come from this.

_Originally posted by @stinodego in https://github.com/pola-rs/polars/issues/15215#issuecomment-2056708405_

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #16158 

This should probably give a performance boost for relevant code, but I didn't check.

Things I didn't touch:

- PyArrow usages (logical_plan/pyarrow.rs), because I didn't understand what was going on there
- Calls from Python code to the old Python functions. These could in theory be switched to calling into Rust...
- Conversions that were deferred to #16199 due to Python 3.9 probably being required to do so.

### CODE REVIEW

File: py-polars/src/conversion/any_value.rs [hunk 1]
```diff
@@ -202,36 +210,23 @@ pub(crate) fn py_object_to_any_value<'py>(
...
-            } else {
-                // This should be faster than calling `timedelta_to_int` again with `"ms"` input.
-                let v_us = py_int.extract::<i128>().unwrap();
-                let v = (v_us / 1000) as i64;
-                AnyValue::Duration(v, TimeUnit::Milliseconds)
-            };
-            Ok(av)
-        })
+        let timedelta = ob.extract::<TimeDelta>()?;
+        if let Some(micros) = timedelta.num_microseconds() {
```
Comment (itamarst) [inline py-polars/src/conversion/any_value.rs:214]: Tried nanoseconds here, it broke the tests, so stuck to micro/milliseconds only.

**Comment by ritchie46** [comment]
Thanks! Can you rebase?

**Comment by itamarst** [comment]
@ritchie46 I fixed the conflict. If you literally want a rebase I can do that too.

**Comment by ritchie46** [comment]
Nope, the conflict is fine. ^^

**Pull request merged**